.github/workflows/build-and-test.yml

```diff
@@ -32,7 +32,12 @@ jobs:
         id: build-orc-orc_dogfood
         continue-on-error: true
         run: |
-          xcodebuild -project ./build/orc.xcodeproj -scheme orc_dogfood -configuration Debug
+          ORC_OUTPUT_FILE=./output.json ORC_OUTPUT_FILE_MODE=json xcodebuild -json -project ./build/orc.xcodeproj -scheme orc_dogfood -configuration Debug -hideShellScriptEnvironment
```

Comment:
A new test that leverages the environment variables to cause a JSON file to get dumped when running `orc_dogfood` (which is a CMake target that uses ORC to _build ORC_). The JSON is then traversed with a couple `jq` queries to make sure some boilerplate keys are in there and filled in with something meaningful.